### ISSUE TITLE
Parenthesis warnings with synchronous code

### ISSUE THREAD

**Issue by marioortizmanero** [fMeow/maybe-async-rs#1]
Hello! First of all, thanks a lot for creating this crate! It's super useful, we're using it at https://github.com/ramsayleung/rspotify.

The only problem we've found is that when `maybe-async/is_sync` is activated, these warnings appear:

```
warning: unnecessary parentheses around block return value
  --> src/lib.rs:11:12
   |
11 |     base().await
   |            ^^^^^ help: remove these parentheses
   |
   = note: `#[warn(unused_parens)]` on by default
```

Here's a minimal snippet:

```rust
//! lib.rs
use maybe_async::{sync_impl, maybe_async};

#[sync_impl]
fn base() -> ureq::Response {
  ureq::get("https://myapi.example.com/ingest")
      .call()
}

#[maybe_async]
pub async fn current_user() -> ureq::Response {
    base().await
}
```

```toml
# Cargo.toml
[package]
name = "notworking"
version = "0.1.0"
authors = ["Mario Ortiz Manero <marioortizmanero@gmail.com>"]
edition = "2018"

[dependencies]
async-trait = "0.1.40"
maybe-async = "0.2.0"
ureq = "1.4.1"

[features]
default = ["maybe-async/is_sync"]
```

Any idea of what this could be? Thanks in advance.

**Comment by fMeow** [comment]
It is hard to perfectly tackle this problem. So I choose the tricky way, that is to allow unused parenthesis for codes converted to sync implementation. I think it reason because only codes that need to be converted to sync are enclosed by the `#[allow(unused_paren)]` macro.

This change is published in v0.2.1.

**Comment by marioortizmanero** [comment]
I've run a `cargo expand` just to see what it's like in the new version:

```rust
#[maybe_async::maybe_async]
trait Trait {
    fn sync_fn() {}

    async fn declare_async(&self);

    async fn async_fn(&self) {
        self.declare_async().await
    }
}

struct Struct;

#[maybe_async::maybe_async]
impl Trait for Struct {
    fn sync_fn() {}

    async fn declare_async(&self) {}

    async fn async_fn(&self) {
        async { self.declare_async().await }.await
    }
}

#[async_std::main]
async fn main() {
    let s = Struct {};
    s.async_fn();
}
```

becomes:

```rust
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::v1::*;
#[macro_use]
extern crate std;
#[allow(unused_parens)]
trait Trait {
    fn sync_fn() {}
    fn declare_async(&self);
    fn async_fn(&self) {
        (self.declare_async())
    }
}
struct Struct;
#[allow(unused_parens)]
impl Trait for Struct {
    fn sync_fn() {}
    fn declare_async(&self) {}
    fn async_fn(&self) {
        #[allow(unused_parens)]
        ({ (self.declare_async()) })
    }
}
fn main() {
    async fn main() {
        {
            let s = Struct {};
            s.async_fn();
        }
    }
    async_std::task::block_on(async { main().await })
}
```

So the implementation makes it hard to avoid these parenthesis? Using `#[allow(unused_parens)]` does the trick but it could avoid actual warnings regarding unnecessary parenthesis inside the rest of the code. For example, if you wrapped the `self.declare_async()` into `(self.declare_async())`, you wouldn't get a warning. I still prefer it this way, but maybe there's an even better solution.

**Comment by fMeow** [comment]
You are right. In the v0.2.2, I replace a paren with block and restricted the span of `#allow(unused_brace)` to the exact await code block replaced by a block.

**Comment by fMeow** [comment]
Sorry for the long wait. I have finally find a method to get rid of all redundant parenthesises and braces.

The new changes have been published to v0.2.2 in crates.io. Have a try please, thank you so much for your feedback! :)
